sdk/identity/identity-cache-persistence/README.md

```diff
@@ -0,0 +1,92 @@
+
+## Getting started
+
+### Install the package
+
+This package is designed to be used with Azure Identity for JavaScript. Install both `@azure/identity` and this package using `npm`:
+
+```sh
+$ npm install --save @azure/identity
+$ npm install --save @azure/identity-cache-persistence
+```
+
+### Prerequisites
+
+- An [Azure subscription](https://azure.microsoft.com/free/).
+
+#### Supported Environments
+
+Azure Identity extensions for JavaScript support stable (even numbered) versions of Node.js starting from v12. While it may run in other Node versions, no support is guaranteed. `@azure/identity-cache-persistence` **does not** support browser environments.
+
+## Key concepts
+
+If this is your first time using `@azure/identity` or the Microsoft identity platform (Azure Active Directory), we recommend that you read [Using `@azure/identity` with Microsoft Identity Platform](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/using-azure-identity.md) first. This document will give you a deeper understanding of the platform and how to configure your Azure account correctly.
+
+### Azure Identity Extensions
+
+As of `@azure/identity` version 2.0.0, the Identity client library for JavaScript includes an extension API. This package (`@azure/identity-cache-persistence`) exports an extension object that you must pass as an argument to the top-level `useIdentityExtension` function from the `@azure/identity` package. Enable token cache persistence in your program as follows:
+
+```typescript
+import { useIdentityExtension } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+```
+
+After calling `useIdentityExtension`, the persistent token cache extension is registered to the `@azure/identity` package and will be available on all credentials that support persistent token caching (those that have `tokenCachePersistenceOptions` in their constructor options).
+
+## Examples
+
+Once the extension is registered, you can enable token cache persistence by passing `tokenCachePersistenceOptions` with an `enabled` property set to `true` to a credential constructor. In the following example, we use the `DeviceCodeCredential`, since persistent caching of its tokens allows you to skip the interactive device-code authentication flow if a cached token is available.
+
+```typescript
+import { useIdentityExtension, DeviceCodeCredential } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+async function main() {
+  const credential = new DeviceCodeCredential({
+    tokenCachePersistenceOptions: {
+      enabled: true
+    }
+  });
+
+  // This will print a JWT access_token and its expiration timestamp
+  console.log("Token:", await credential.getToken());
```

Comment:
to avoid folks copy-pasting themselves into revealing secrets, should we maybe truncate this string with substr or somesuch?